proxy/http2/Http2ConnectionState.cc

```diff
@@ -947,6 +938,11 @@ Http2ConnectionState::state_closed(int /* event */, void * /* edata */)
 Http2Stream *
 Http2ConnectionState::create_stream(Http2StreamId new_id, Http2Error &error)
 {
+  // In half_close state, TS doesn't create new stream. Because GOAWAY frame is sent to client
+  if (ua_session && ua_session->get_half_close_flag()) {
+    return nullptr;
```

Comment:
How about `(Http2ErrorClass::HTTP2_ERROR_CLASS_STREAM, Http2ErrorCode::HTTP2_ERROR_REFUSED_STREAM)`? TS will send RST_STREAM and client can retry the request safely.